src/main/java/com/mongodb/hibernate/internal/service/StandardServiceRegistryScopedState.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.internal.service;
+
+import static com.mongodb.hibernate.internal.VisibleForTesting.AccessModifier.PRIVATE;
+import static java.lang.String.format;
+import static org.hibernate.cfg.AvailableSettings.JAKARTA_JDBC_URL;
+
+import com.mongodb.hibernate.internal.VisibleForTesting;
+import com.mongodb.hibernate.internal.cfg.MongoConfiguration;
+import com.mongodb.hibernate.internal.cfg.MongoConfigurationBuilder;
+import com.mongodb.hibernate.service.MongoConfigurationContributor;
+import java.io.Serial;
+import java.util.Map;
+import org.hibernate.HibernateException;
+import org.hibernate.boot.registry.StandardServiceInitiator;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.service.Service;
+import org.hibernate.service.UnknownServiceException;
+import org.hibernate.service.spi.ServiceRegistryImplementor;
+import org.jspecify.annotations.Nullable;
+
+public final class StandardServiceRegistryScopedState implements Service {
```

Comment:
We will be able to use the single instance of this service as shared state between the MQL translation code (plugged via `MongoDialect`) the JDBC adapter code (plugged via `MongoConnectionProvider`). This is the solution to the "we can't use static state, if we want to support multiple session factories per application" issue.